core/src/main/scala/org/http4s/EntityDecoder.scala

```diff
@@ -177,4 +177,11 @@ trait EntityDecoderInstances {
 
   implicit def multipart: EntityDecoder[Multipart] =
     MultipartDecoder.decoder
+
+  /** An entity decoder that ignores the content and returns unit. */
+  implicit val void: EntityDecoder[Unit] = new EntityDecoder {
+    val consumes: Set[MediaRange] = Set.empty
+
+    def decode(msg: Message, strict: Boolean): DecodeResult[Unit] = DecodeResult.success(())
```

Comment:
This is nice for http/1.x so as to preserve the session, but it's wasteful in http/2. Currently most (all?) of our usage is http/1.x, so it may still be the best thing to do for now, but I'd hate to change it later in life and surprise folks.
I'm not sure what's the best thing to do here.